resotoclient/__init__.py

```diff
@@ -139,40 +129,29 @@ def _post(
         params: Optional[Dict[str, str]] = None,
         headers: Optional[Dict[str, str]] = None,
         stream: bool = False,
-    ) -> requests.Response:
-        with requests.Session() as s:
-            self._prepare_session(s)
-            if stream:
-                s.stream = True
-                s.headers.update({"Accept": "application/ndjson"})
-            if headers:
-                headers.update(headers)
-            return s.post(
-                self.resotocore_url + path,
-                data=data,
-                json=json,
-                params=params,
-                headers=headers,
-            )
+    ) -> HttpResponse:
+        if not self.sync_client.running():
+            self.sync_client.start()
+        return self.sync_client.post(path, json, data, params, headers, stream)
 
     def _put(
         self, path: str, json: JsValue, params: Optional[Dict[str, str]] = None
-    ) -> requests.Response:
-        with requests.Session() as s:
-            self._prepare_session(s)
-            return s.put(self.resotocore_url + path, json=json, params=params)
+    ) -> HttpResponse:
+        if not self.sync_client.running():
+            self.sync_client.start()
+        return self.sync_client.put(path, json, params)
 
-    def _patch(self, path: str, json: JsValue) -> requests.Response:
-        with requests.Session() as s:
-            self._prepare_session(s)
-            return s.patch(self.resotocore_url + path, json=json)
+    def _patch(self, path: str, json: JsValue) -> HttpResponse:
+        if not self.sync_client.running():
+            self.sync_client.start()
+        return self.sync_client.patch(path, json)
 
     def _delete(
         self, path: str, params: Optional[Dict[str, str]] = None
-    ) -> requests.Response:
-        with requests.Session() as s:
-            self._prepare_session(s)
-            return s.delete(self.resotocore_url + path, params=params)
+    ) -> HttpResponse:
+        if not self.sync_client.running():
```

Comment:
Saw this code so often. Should this be a `@requires_sync_client_running` decorator or maybe `start()` includes the running check so the if could be saved or something along those lines?